common/helpers/markers/src/markers/__init__.py

```diff
@@ -42,11 +47,19 @@ def create_and_publish_marker(
     r: float = 0.0,
     g: float = 1.0,
     b: float = 0.0,
+    name: Optional[str] = None,
 ):
     if idx is None:
         global publisher_counts
         idx = publisher_counts[publisher]
         publisher_counts[publisher] += 1
-
     marker_msg = create_marker(point_stamped, idx, r, g, b)
+    idx = publisher_counts[publisher]
+    publisher_counts[publisher] += 1
     publisher.publish(marker_msg)
+    rospy.sleep(2)  # Needed to prevent markers from being overwritten
+    if name is not None:
+        name_location = point_stamped.point
+        name_location.z += 0.1
+        marker_name_msg = create_marker(point_stamped, idx, r, g, b, name)
+        publisher.publish(marker_name_msg)
```

Comment:
```suggestion
    publisher.publish(marker_msg)
    rospy.sleep(2)  # Needed to prevent markers from being overwritten
    if name is not None:
        name_location = point_stamped.point
        name_location.z += 0.1
        idx = publisher_counts[publisher]
        publisher_counts[publisher] += 1
        marker_name_msg = create_marker(point_stamped, idx, r, g, b, name)
        publisher.publish(marker_name_msg)
```